server/pkg/asset/asset.go

```diff
@@ -45,13 +47,20 @@ func (a *Asset) Size() uint64 {
 }
 
 func (a *Asset) PreviewType() *PreviewType {
+	if a.previewType == nil {
+		return lo.ToPtr(PreviewType(""))
+	}
 	return a.previewType
 }
 
 func (a *Asset) File() *File {
 	return a.file
 }
 
-func (a *Asset) Hash() string {
-	return a.hash
+func (a *Asset) UUID() string {
+	return a.uuid
+}
+
+func (a *Asset) UpdatePreviewType(p *PreviewType) {
+	a.previewType = p
```

Comment:
```suggestion
	a.previewType = util.CloneRef(p)
```

Don't just imitate them superficially, but think about why you do it:

```go
pt := lo.ToPtr(PreviewTypeHoge)
asset.UpdatePreviewType(pt)
*pt = PreviewTypeAnother
assert.NotEqual(t, PreviewTypeHoge, *asset.PreviewType())
// The inner field was unexpectedly changed. It means we cannot force developers to force using UpdatePreviewType to update the preview type.
```